CentralUnit/Communication/XbeeNetwork.cs

```diff
@@ -346,22 +357,33 @@ private void ProcessXbeePacket(XBeePacket packet)
         {
             if (packet != null)
             {
+                if (Log.IsDebugEnabled)
+                {
+                    Log.Debug(packet.ToPrettyString());
+                }
+
                 if (packet is ExplicitRxIndicatorPacket eriPacket)
                 {
                     XbeeSerialCommunication deviceForPacket = this.GetDevice(eriPacket.SourceAddress64);
                     deviceForPacket.OnDataReceived(eriPacket.RFData);
                 }
+                else if (packet is ReceivePacket recPacket)
+                {
+                    XbeeSerialCommunication deviceForPacket = this.GetDevice(recPacket.SourceAddress64);
+                    deviceForPacket.OnDataReceived(recPacket.RFData);
+
+                }
                 else if (packet is TransmitStatusPacket txsPacket)
                 {
                     this.ProcessTransmitStatusPacket(txsPacket);
                 }
+                else if (packet is ATCommandResponsePacket atrPacket)
+                {
+                    Log.Info($"Got response for AT command {atrPacket.StringCommandValue}, status {atrPacket.Status}");
+                }
                 else
                 {
-                    Log.Info($"Got unknown packet of type {packet.GetType().Name}");
-                    if (Log.IsDebugEnabled)
-                    {
-                        Log.Debug(packet.ToPrettyString());
-                    }
+                    Log.Info($"Got unknown packet of type {packet.GetType().Name}");   
```

Comment:
switch syntax also supports pattern matching like `packet is TransmitStatusPacket txsPacket` so i think thisif/else block is big ewnough to become a switch

Reply:
Wasn't aware of this. Will refactor